### ISSUE TITLE
1337 - Incorrect Mint Fee Calculation in mintBatch Function Allows Reduced Cost Token Minting

### ISSUE THREAD

**Issue by sherlock-admin4** [sherlock-audit/2024-04-titles-judging#83]
1337

high

# Incorrect Mint Fee Calculation in mintBatch Function Allows Reduced Cost Token Minting

## Summary
The mintBatch function in Edition.sol incorrectly calculates the mint fee, allowing users to mint tokens at a reduced cost by exploiting the batch minting functionality.

## Vulnerability Detail
The vulnerability lies in the mintBatch function, which allows minting tokens to multiple receivers in a single transaction. The function calls FEE_MANAGER.collectMintFee only once, passing the amount_ parameter as the amount of tokens to mint. However, inside the loop, the _issue function is called for each receiver, effectively minting amount_ tokens to each receiver.

As a result, the total number of tokens minted is `amount_ * receivers_.length`, while the mint fee is collected only for amount_ tokens. This discrepancy allows users to mint a larger quantity of tokens than intended while paying a reduced mint fee.

## Impact

The impact of this vulnerability is that users can exploit the mintBatch function to mint tokens at a lower cost than intended. By providing a large number of receivers and a relatively small amount_ value, users can mint a significant number of tokens while paying a disproportionately low mint fee.

## Code Snippet
```solidity
    /// @notice Mint a token to a set of receivers for the given work.
    /// @param receivers_ The addresses to mint the tokens to.
    /// @param tokenId_ The ID of the work to mint.
    /// @param amount_ The amount of tokens to mint.
    /// @param data_ The data associated with the mint. Reserved for future use.
    function mintBatch(
        address[] calldata receivers_,
        uint256 tokenId_,
        uint256 amount_,
        bytes calldata data_
    ) external payable {
        // wake-disable-next-line reentrancy
        FEE_MANAGER.collectMintFee{value: msg.value}(
            this, tokenId_, amount_, msg.sender, address(0), works[tokenId_].strategy
        );

        for (uint256 i = 0; i < receivers_.length; i++) {
            _issue(receivers_[i], tokenId_, amount_, data_);
        }

        _refundExcess();
    }
```
https://github.com/sherlock-audit/2024-04-titles/blob/main/wallflower-contract-v2/src/editions/Edition.sol#L304
## Tool used

Manual Review

## Recommendation
To fix this vulnerability, the mintBatch function should calculate the total amount of tokens to be minted based on the number of receivers and pass that value to FEE_MANAGER.collectMintFee. 

Duplicate of #264
